### DISCUSSION TITLE
Hardcoding UART ports for Modbus Serial is very limiting

### DISCUSSION THREAD

**Discussion by LucasSaintrain** [espressif/esp-modbus#64] [Ideas]
Modbus is an application protocol that defines rules for messaging structure and data organization that are independent of the data transmission medium. Hardcoding calls to the UART driver inside the library limits the use case for other types of serial communication, notably Bluetooth SPP.

**Comment by alisitsyn** [comment]
@LucasSaintrain, 

You are right about Modbus protocol and its independence from transmission media. Please note that Modbus standard clarifies just two typical physical layers such as serial ( EIA/TIA -232-E, EIA-422, EIA/TIA-485-A ) and  TCP/IP over Ethernet. These layers were included as standard and typical use causes and added to requirement list. However, the interface for Master and Slave allows to add additional objects (controller, port) to handle other media layers into stack but reuse the common functionality of other layers for Modbus.
Your request for BlueTooth SPP is added to the feature list for realization together with other ideas.
